### DISCUSSION TITLE
Modrule with ApisixRoute Object

### DISCUSSION THREAD

**Discussion by ltehg** [kubemod/kubemod#121] [Q&A]
Hello, 

I’m experimenting with Kubemod and  have a problem with apisixroute that the modrule does not seem to work here.
The modrule on Deployment works perfectly. 
But if kind = ApisixRoute  the object apisixroute is still created in the development

Here is a simple example.  

rule.yaml:

apiVersion: api.kubemod.io/v1beta1
kind: ModRule
metadata:
  name: reject-apisixroute
  namespace: development
spec:
  type: Reject
  match:
    - select: '$.kind'
      matchValue: 'ApisixRoute'
  rejectMessage: " Rejected ApisixRoute!"

route.yaml:

apiVersion: apisix.apache.org/v2
kind: ApisixRoute
metadata:
  name: example-route
  namespace: development
spec:
  http:
    - name: test-rule
      match:
        hosts:
          - test.com
        paths:
          - /*
      authentication:
        enable: true
        type: keyAuth
      backends:
        - serviceName: example-service
          servicePort: 8080

can someone help ? it is something not correct ? 
regards lt

**Comment by ltehg** [comment]
The solution is to have patching the webhook!   
https://github.com/kubemod/kubemod?tab=readme-ov-file#patch-section

**Comment by vassilvk** [comment]
Did you make sure to add the `apisixroutes` resource to the target resources as per [this section](https://github.com/kubemod/kubemod?tab=readme-ov-file#target-resources)?

**Reply by ltehg** [reply]
Yes the solution for me is add apisixroute to target resources. 
We are very positive about kubemod so far and will continue to test it. Thx
